packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -550,7 +550,9 @@ class _SnackBarState extends State<SnackBar> {
         child: snackBar,
       );
     }
-
-    return ClipRect(child: snackBarTransition);
+    return Hero(
```

Comment:
This looks slick. But what happens if you write
```dart
Theme(
  data: ThemeData(snackBarTheme: mySnackBarTheme),
  child: ... showSnackBar(mySnackBar),
)
```
We may need to explain this limitation.

Reply:
That wouldn't really warn anyone about this case (unless they do a lot of recreational reading). On the other hand, this (Hero) issue doesn't come up that often, which is why it's not called out in the Hero docs. Hopefully it will not come up often due to snackbar heroics either.